catalyst/metrics/cmc_score.py

```diff
@@ -56,4 +56,45 @@ def cmc_score(
     return cmc_score_count(distances, conformity_matrix, topk)
 
 
-__all__ = ["cmc_score_count", "cmc_score"]
+def masked_cmc_score(
+    query_embeddings: torch.Tensor,
+    gallery_embeddings: torch.Tensor,
+    conformity_matrix: torch.Tensor,
+    available_samples: torch.Tensor,
+    topk: int = 1,
+) -> float:
+    """
+
+    Args:
+        query_embeddings: tensor shape of (n_embeddings, embedding_dim)
+            embeddings of the objects in query
+        gallery_embeddings: tensor shape of (n_embeddings, embedding_dim)
+            embeddings of the objects in gallery
+        conformity_matrix: binary matrix with 1 on same label pos
+            and 0 otherwise
+        available_samples: tensor of shape (query_size, gallery_size),
+            available_samples[i][j] == 1 means that j-th element of gallery
+            should be used while scoring i-th query one
+        topk: number of top examples for cumulative score counting
+
+    Returns:
+        cmc score with mask
+
+    Raises:
+        ValueError: if there are items that have different labels and are
+            unavailable for each other according to availability matrix
+    """
+    if not available_samples[conformity_matrix == 0].all():
+        raise ValueError(
+            "In reid case we suppose that all the pids that are not "
+            "the same with query pid should be available while model "
```

Comment:
this comment is confusing a little bit. did you want to say 'the same query CID'? hmm

Reply:
we should polish it